src/main/java/com/microsoft/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -0,0 +1,104 @@
+package com.microsoft.cse.helium.app.services.keyvault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Value;
+import java.io.IOException;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.MSICredentials;
+import com.microsoft.azure.credentials.AzureCliCredentials;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.models.CertificateBundle;
+import com.microsoft.azure.keyvault.models.KeyBundle;
+import com.microsoft.azure.keyvault.models.SecretBundle;
+
+@Service
+public class KeyVaultService implements IKeyVaultService
+{
+    private String _keyVaultName = "";
+    private String _environmentFlag = "";
+    private AzureTokenCredentials _credentials;
+    private KeyVaultClient _keyVaultClient;
+
+    private final String _keyVaultNameRegex ="^[a-zA-Z](?!.*--)([a-zA-Z0-9-]*[a-zA-Z0-9])?$";
+    private static final Logger _logger = LoggerFactory.getLogger(KeyVaultService.class);
+
+    public KeyVaultService (@Value("${helium.keyvault.name}") String keyVaultName, @Value("${helium.environment.flag}") String environmentFlag) throws Exception {
```

Comment:
KeyVaultName is the standard env var

Reply:
It uses KeyVaultName, look in application.properties you'll see helium.keyvault.name=${KeyVaultName}.